cmd/tests/cmd_run_test.go

```diff
@@ -1998,3 +2000,99 @@ func TestBadLogOutput(t *testing.T) {
 		})
 	}
 }
+
+// Tests that the appropriate events are emitted at the appropriate times.
+func TestEventSystemOK(t *testing.T) {
+	t.Parallel()
+
+	ts := NewGlobalTestState(t)
+
+	// HACK: We need to be able to run multiple tests in parallel on the events
+	// module, but since it does event subscription in a sync.Once and the API
+	// addresses need to be different for separate tests, we register separate
+	// modules to workaround it. I experimented with passing the address via
+	// the script, but the sync.Once is still a problem.
+	// Note that this also forbids running these tests with more than -count=1,
+	// unless we dynamically increment the module name... /sigh
+	modules.Register("k6/x/testevents1", events.New(
+		ts.GlobalState.DefaultFlags.Address, []event.Type{
+			event.Init, event.TestStart, event.IterStart, event.IterEnd,
+			event.TestEnd, event.Exit,
+		}))
```

Comment:
Any ideas for improving this?

Reply:
Since some of the last changes in `js` package the module system gets teh modules it should support 

https://github.com/grafana/k6/blob/cf90835d199e8f9c89e9621f33a6e655dab94b04/js/bundle.go#L110

This is not exposed so that you can create a runner wtih *specific* modules. 

At this point I feel like this is not a bad idea, but also that likely we should do some moving around of the current `modules` registration system ... once again.

I don't really like how one of the modules are in one place and the other in another and are being merged strangely.

A quicker fix might be for the Runner to get only the "non-standard"/extension modules. But again this likely will be too big of a change.

I am mostly writing to ask you to actually do teh hack with the ever increasing index so that `-count 2` can run as I always find it very strange and frustrating when tests suddenly stop working with it and I am using it for some bug/race teasing.

An example of the hack: (whcih I think I can remove in that place :sweat_smile: ) 

https://github.com/grafana/k6/blob/cf90835d199e8f9c89e9621f33a6e655dab94b04/js/init_and_modules_test.go#L38-L44

Reply:
> So we need to return a local map, similarly as in getInternalJSModules(), but first we should get rid of the registration in init() pattern all extensions use, which is at the crux of the issue

the `init()` pattern has nothing to do with this. 

Even if we were using something else I doubt the js.Runner would've been written to get it from outside back then to begin with. If it is not taking it from the outside - it will still get it in some "global" way that will mean that you won't control it. 

And that is the problem. Removign `init()` will just break all extensions and will (IMO) make it more likely we will break them again and again in the future.

The curretn solution that I see that won't be that bad is for `js.Runner#New` (and NewArchive) to take `map[string]interface{}` with the custom modules and propage it through the code. Instead of it being taken from `getInternalJSModules()`


The rest of the comment - I don't understand:
1. at a call to `New()` or `NewModulesInstnace()` the options are not completed - so getting access to them is pointless, it won't help you in most cases.
2. `since you were against changing New()` that is because it is not part of the API. It is just a convention. And at the time it is called there is nothing you can do with it. If you want to make it part of the API you will basically make a completely new API.